operator/watchers/envoy_config.go

```diff
@@ -0,0 +1,216 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2019-2020 Authors of Cilium
+
+package watchers
+
+import (
+	"fmt"
+
+	"github.com/cilium/cilium/pkg/envoy"
+	"github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	slim_networkingv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/networking/v1"
+	envoy_config_cluster_v3 "github.com/cilium/proxy/go/envoy/config/cluster/v3"
+	envoy_config_listener "github.com/cilium/proxy/go/envoy/config/listener/v3"
+	envoy_config_route_v3 "github.com/cilium/proxy/go/envoy/config/route/v3"
+	envoy_extensions_filters_network_http_connection_manager_v3 "github.com/cilium/proxy/go/envoy/extensions/filters/network/http_connection_manager/v3"
+	"github.com/golang/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+	v1meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func amazingIngressControllerBusinessLogic(ingress *slim_networkingv1.Ingress) (*v2alpha1.CiliumEnvoyConfig, error) {
+	backendServices := getBackendServices(ingress)
+	resources, err := getResources(ingress, backendServices)
+	if err != nil {
+		return nil, err
+	}
+	return &v2alpha1.CiliumEnvoyConfig{
+		TypeMeta: v1meta.TypeMeta{
+			Kind:       v2alpha1.CECKindDefinition,
+			APIVersion: "cilium.io/v2alpha1",
+		},
+		ObjectMeta: v1meta.ObjectMeta{
+			Name: ingress.Name,
+		},
+		Spec: v2alpha1.CiliumEnvoyConfigSpec{
+			Services: []*v2alpha1.ServiceListener{
+				{
+					Name:      getServiceNameForIngress(ingress),
+					Namespace: ingress.Namespace,
+					Listener:  ingress.Name,
+				},
+			},
+			BackendServices: backendServices,
+			// TODO(michi) what is this
+			// Ingress:         true,
```

Comment:
This is a flag for *proxy ingress to pod* vs. *proxy egress from pod*. This is not to be used with k8s Ingress :-). Also, it is yet to be seen if we will ever actually need CiliumEnvoyConfig for pod ingress. So far it is completely untested and the bpf datapath support for it is also missing.